### PR TITLE
Add missing env variables to template.

### PR DESCRIPTION
**Pull request by dtaskai**
*Description of changes:* Add missing environment variables to Athena DynamoDB connector template, according to documentation: https://docs.aws.amazon.com/athena/latest/ug/connectors-dynamodb.html

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by dtaskai** [comment]
We dearly miss these few environment variables from the template, as without them being present in the template we have no way of setting them programmatically in CDK, only manually on the Lambda itself. 😞

File: athena-dynamodb/athena-dynamodb.yaml [hunk 1]
```diff
@@ -24,6 +24,20 @@ Parameters:
     Description: 'The prefix within SpillBucket where this function can spill data.'
     Type: String
     Default: athena-spill
+  SpillPutRequestHeaders:
+    Description: '(Optional) A JSON encoded map of request headers and values for the Amazon S3 putObject request that is used for spilling. (for example, {"x-amz-server-side-encryption" : "aws:kms"}).'
+    Type: String
+  DisableGlue:
+    Description: "(Optional) If present and set to 'true', the connector does not attempt to retrieve supplemental metadata from AWS Glue."
+    Default: 'false'
+    Type: String
```
Comment (chngpe) [inline athena-dynamodb/athena-dynamodb.yaml:33]: Add allowedValues to provided validation on input 
    AllowedValues:
      - true
      - false

File: athena-dynamodb/athena-dynamodb.yaml [hunk 2]
```diff
@@ -24,6 +24,20 @@ Parameters:
...
+  GlueCatalog:
+    Description: '(Optional) Use this option to specify the account id of a different account for a cross-account AWS Glue catalog. By default, the connector attempts to get metadata from its own AWS Glue account.'
+    Type: String
+  DisableProjectionAndCasing:
+    Description: "(Optional) Disables projection and casing. Use if you want to query DynamoDB tables that have casing in their column names and you do not want to specify a columnMapping property on your AWS Glue table. Can be set to 'auto' or 'always', set to 'auto' by default. See documentation for more information."
+    Default: 'auto'
+    Type: String
```
Comment (chngpe) [inline athena-dynamodb/athena-dynamodb.yaml:40]: Add allowedValues to provided validation on input
    AllowedValues:
      - auto
      - always

File: athena-dynamodb/athena-dynamodb.yaml [hunk 3]
```diff
@@ -51,6 +65,8 @@ Parameters:
 
 Conditions:
   HasKMSKeyId: !Not [!Equals [!Ref KMSKeyId, ""]]
+  HasSpillPutRequestHeaders: !Not [!Equals [!Ref HasSpillPutRequestHeaders, ""]]
```
Comment (chngpe) [inline athena-dynamodb/athena-dynamodb.yaml:68]: This `HasSpillPutRequestHeaders` should be `SpillPutRequestHeaders`

File: athena-dynamodb/athena-dynamodb.yaml [hunk 4]
```diff
@@ -24,6 +24,20 @@ Parameters:
...
+  SpillPutRequestHeaders:
+    Description: '(Optional) A JSON encoded map of request headers and values for the Amazon S3 putObject request that is used for spilling. (for example, {"x-amz-server-side-encryption" : "aws:kms"}).'
+    Type: String
+  DisableGlue:
+    Description: "(Optional) If present and set to 'true', the connector does not attempt to retrieve supplemental metadata from AWS Glue."
+    Default: 'false'
+    Type: String
+  GlueCatalog:
+    Description: '(Optional) Use this option to specify the account id of a different account for a cross-account AWS Glue catalog. By default, the connector attempts to get metadata from its own AWS Glue account.'
+    Type: String
```
Comment (chngpe) [inline athena-dynamodb/athena-dynamodb.yaml:36]: allowed Values as `'^\d{12}$'`

**Comment by chngpe** [comment]
Hi!

Thank you for submitting the PR for adding the missing environment variables into the template. 
I left some comments, mostly on restrict the user input to only the allowed values. 

-P

**Comment by henrymai** [comment]
This PR shouldn't actually be necessary if all you need to do is set environment variables after deployment with CDK.

See my other comment here about the same topic: https://github.com/awslabs/aws-athena-query-federation/issues/1073#issuecomment-1445026825

**Comment by dtaskai** [comment]
Thanks for the input and the proposed solution @henrymai! though I feel like the expected behavior would be that all of the documented environment variables are available in the template, not just select ones (also makes for a much leaner solution in CDK), that is why we were surprised that our deployment was failing in the first place.

**Comment by atipapp** [comment]
@henrymai seems like Dominik addressed your comment and I agree with him. I think specifying it through a template parameter would be leaner.

 @chngpe can you please re-review?

**Comment by henrymai** [comment]
@atipapp I don't work on this team anymore. I just added my comment as a way for you guys to resolve your problems faster without waiting for another release.

**Comment by chngpe** [comment]
Hi! 
Sorry for late response.

After discussion this internally with team. The parameters was intentionally left out in the template. We want the templates to be extremely light weight and easier for customer to get started. Unfortunately, due to the same reason team decide not to add more parameters to the template. If needed, please follow the suggestion mentioned above

```
https://github.com/awslabs/aws-athena-query-federation/pull/1394#issuecomment-1653241263
https://github.com/awslabs/aws-athena-query-federation/issues/1073#issuecomment-1445026825
```

**Pull request closed without merging**